File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
import java.util.Arrays;

class ResistorColorDuo {

    static public final String[] COLORS = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};

    int value(String[] colors) {
        return Arrays.asList(COLORS).indexOf(colors[0]) * 10 + Arrays.asList(COLORS).indexOf(colors[1]);
    }
}

File: solutions/ruby/resistor-color-duo/1/resistor_color_duo.rb
module ResistorColorDuo
  COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
  def self.value(colorPair)
    "#{COLORS.index(colorPair[0])}#{COLORS.index(colorPair[1])}".to_i
  end
end

